pkg/hubble/observer/types/types.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2020 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package types
+
+import "time"
+
+const (
+	// LostEventSourceUnspec indicates an event has been lost at an unknown
+	// source
+	LostEventSourceUnspec = iota
+	// LostEventSourcePerfRingBuffer indicates an event has been lost because
+	// the perf event ring buffer was not read before it was overwritten.
+	LostEventSourcePerfRingBuffer
+	// LostEventSourceEventsQueue indicates that an event has been dropped
+	// because the events queue was full.
+	LostEventSourceEventsQueue
+)
+
+// MonitorEvent is the top-level type for all events consumed by the observer
+type MonitorEvent struct {
+	// Timestamp when the event was received by the consumer
+	Timestamp time.Time
+	// NodeName where the event occurred
+	NodeName string
+	// Payload is one of: AgentEvent, PerfEvent or LostEvent
```

Comment:
should we use proto to assist with these structs?

Reply:
I'm a bit hesitant to add proto here, it would be a leaky abstraction of the proto API. The distinction between a perf event (datpath) and agent event (controlpath) makes sense in the context of a monitor subscriber, but I don't think it's a distinction we want to carry over to the external API exposed to users. 

Plus them being internal types allows us to be more flexible with the interface, not locking is down for future refactors (which is the reason I had to introduce a new interface in the first place, because the old monitor `payload.Payload` was designed to be written in a unix domain socket, thus only supporting data of type `[]byte`)